samples/python/console/speech_sample.py

```diff
@@ -700,7 +700,7 @@ def recognized(evt):
     # For continuous pronunciation assessment mode, the service won't return the words with `Insertion` or `Omission` even miscue is enabled.
     # We need to compare with the reference text after received all recognized words to get these error words.
     if enable_miscue:
-        reference_words = reference_text.lower().split().strip(string.punctuation)
+        reference_words = [w.strip(string.punctuation) for w in reference_text.lower().split()]
```

Comment:
>[w.strip(string.punctuation) for w in reference_text.lower().split()] [](start = 26, length = 69)

This correlate query style of removing the punctuation is amazing, but not easy to follow unless you are familiar with it, or the SQL version of it.

Would it be better for the sample, if we split this into multiple steps that are easy to follow?

Reply:
it's a good comment, I add some comments to explain this line.